llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1506,6 +1506,16 @@ bool LoopVectorizationLegality::canVectorize(bool UseVPlanNativePath) {
       return false;
   }
 
+  if (isa<SCEVCouldNotCompute>(PSE.getBackedgeTakenCount())) {
```

Comment:
What about the other users of LAA in tree?  Have you audited them?  If not, can you add bailouts to ensure we're not breaking anything with this transition?

Reply:
I checked `LoopVersioningLICM`, `LoopDistribute.cpp` & `LoopLoadElimination.cpp`, they all check for a single exiting block, which should be sufficient I think. I also added tests for loop-distribute & loop-load-elim.